tools/const_finder/lib/const_finder.dart

```diff
@@ -32,9 +32,20 @@ class _ConstVisitor extends RecursiveVisitor<void> {
   final List<Map<String, dynamic>> constantInstances;
   final List<Map<String, dynamic>> nonConstantLocations;
 
+  // A cache of previously evaluated classes.
+  static Map<Class, bool> _classHeirarchyCache = <Class, bool>{};
   bool _matches(Class node) {
-    return node.enclosingLibrary.importUri.toString() == classLibraryUri &&
-      node.name == className;
+    if (node == null) {
+      return false;
+    }
+    if (_classHeirarchyCache[node] == true) {
```

Comment:
Caching `false` answer is even more useful (because it is much more common). So consider something like

```dart
  final result = _classHeirarchyCache[node];
  if (result != null) return result;
```

Reply:
Makes sense, done.